accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,162 @@
+            <label #label [id]="idMap['label']">{{stringMap['BLOCK_ACTION_LIST']}}</label>
+            <ol role="group" *ngIf="displayBlockMenu"  [attr.aria-level]="level+2">
+              <li #workspaceCopy [id]="idMap['workspaceCopy']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['workspaceCopyButton'], 'blockly-button')" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button #workspaceCopyButton [id]="idMap['workspaceCopyButton']" 
+                    (click)="copyToWorkspace(block)">{{stringMap['COPY_TO_WORKSPACE']}}</button>
+              </li>
+              <li #blockCopy [id]="idMap['blockCopy']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['blockCopyButton'], 'blockly-button')" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button #blockCopyButton [id]="idMap['blockCopyButton']" 
+                    (click)="clipboardService.copy(block, true)">{{stringMap['COPY_TO_CLIPBOARD']}}</button>
+              </li>
+              <li #sendToSelected [id]="idMap['sendToSelected']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['sendToSelectedButton'], 'blockly-button', utilsService.getMarkedBlockCompatibilityHTMLText(clipboardService.isBlockCompatibleWithMarkedConnection(block)))" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button #sendToSelectedButton [id]="idMap['sendToSelectedButton']" (click)="copyToMarked(block)" 
+                    [disabled]="utilsService.getMarkedBlockCompatibilityHTMLText(clipboardService.isBlockCompatibleWithMarkedConnection(block))">
+                    {{stringMap['COPY_TO_MARKED_SPOT']}}</button>
+              </li>
+            </ol>
+          </li>
+          <div *ngFor="#inputBlock of block.inputList; #i=index">
+            <field-view *ngFor="#field of inputBlock.fieldRow; #j=index" [attr.aria-level]="level+1" [field]="field" [level]="level+1"></field-view>
+            <toolbox-tree-view *ngIf="inputBlock.connection && inputBlock.connection.targetBlock()" [block]="inputBlock.connection.targetBlock()" [displayBlockMenu]="false" [level]="level+1"></toolbox-tree-view>
+            <li #listItem1 [id]="idMap['listItem' + i]" role="treeitem" 
+                *ngIf="inputBlock.connection && !inputBlock.connection.targetBlock()" 
+                [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-argument-text', idMap['listItem' + i + 'Label'])" 
+                [attr.aria-level]="level+1" aria-selected=false>
+              <!-- i18n here will need to happen in a different way due to the way grammar changes based on language. Leaving that as a TODO(sll@) -->
```

Comment:
Also, technically, this should be TODO(madeeha). The LDAP used is the person who wrote the TODO and has the necessary context, in case questions arise. It doesn't refer to person who should fix it.